tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/timeline/timeline_container.ts

```diff
@@ -122,6 +125,18 @@ export class TimelineContainer {
     select(getExecutionScrollBeginIndex)
   );
 
+  readonly scrollBeginIndexUpperLimit$ = this.store.pipe(
+    select(
+      createSelector(
+        getNumExecutions,
+        getDisplayCount,
+        (numExecutions, displayCount) => {
+          return Math.max(0, numExecutions - displayCount);
```

Comment:
I think this invariant has to be enforced by the store (write time). Do we absolutely need one at the read time?

Reply:
There is no invariant here. `numExecutions` can be legitimately `>`, `=`, or `<` `displayCount`. This selector is just determining which of those is the case and help decide whether the slider should be shown or not.